src/audio/module_adapter/module_adapter.c

```diff
@@ -825,6 +840,57 @@ static int module_adapter_simple_copy(struct comp_dev *dev)
 	return ret;
 }
 
+static inline int module_adapter_simple_copy_one_source_one_sink(struct processing_module *mod)
+{
+	struct comp_buffer __sparse_cache *source_c;
+	struct comp_buffer __sparse_cache *sink_c;
+	struct comp_dev *dev = mod->dev;
+	uint32_t frames;
+	int ret = 0;
+
+	source_c = buffer_acquire(mod->source);
+	sink_c = buffer_acquire(mod->sink);
+	frames = audio_stream_avail_frames_aligned(&source_c->stream, &sink_c->stream);
+	mod->input_buffers[0].size = frames;
+	mod->input_buffers[0].consumed = 0;
+	mod->input_buffers[0].data = &source_c->stream;
+	mod->output_buffers[0].size = 0;
+	mod->output_buffers[0].data = &sink_c->stream;
+	if (!mod->skip_src_buffer_invalidate) // add mod->is_multi_core &&
+		buffer_stream_invalidate(source_c,
+					 frames * audio_stream_frame_bytes(&source_c->stream));
+
+	ret = module_process(mod, mod->input_buffers, 1, mod->output_buffers, 1);
+	if (ret) {
+		if (ret != -ENOSPC && ret != -ENODATA) {
+			comp_err(dev, "module_adapter_simple_copy() process failed with error: %x",
+				 ret);
+			goto out;
+		}
+
+		ret = 0;
+	}
+
+	/* consume from input buffer */
+	mod->total_data_consumed += mod->input_buffers[0].consumed;
+	if (mod->input_buffers[0].consumed)
+		audio_stream_consume(&source_c->stream, mod->input_buffers[0].consumed);
```

Comment:
the original version uses `comp_update_buffer_{consume,produce}()` which also calls the notifier. Is this change on purpose?

Reply:
It's using part of Liam's patch. Copied accidentally, but this is better, right?